src/renderer/viz/expressions/basic/property.js

```diff
@@ -63,8 +63,8 @@ export default class Property extends BaseExpression {
         }
         this.type = metaColumn.type;
 
-        if (this.type === 'category') {
-            this.numCategories = metaColumn.categories.length;
+        if (this.type === 'category' && !this.numCategories) {
+            Object.defineProperty(this, 'numCategories', { get: function () { return metaColumn.categories.length; } });
```

Comment:
I have also added a method named `getCategories` that I needed in the `ramp.getLegend` PR https://github.com/CartoDB/carto-vl/pull/869 to get the categories of a property.

I think it would be better to have a `categories` getter and then get the length when needed, or have both a `categories` and `numCategories` getters. It would return the respective values only if the property type is a category, otherwise it would return undefined. I think it is a better approach because defining a property in the constructor is harder to test and read. What do you think?

Reply:
Seems like a good idea, but I would do that on your PR :slightly_smiling_face: